src/meadowflow/scheduler.py

```diff
@@ -221,24 +222,18 @@ def register_job_runner(
         is schedule a callback on the event_loop, so it's possible no state has changed
         when this function returns.
         """
-        self._event_loop.call_soon_threadsafe(
+        asyncio.get_running_loop().call_soon(
             self.register_job_runner_on_event_loop, job_runner_constructor
         )
 
     def register_job_runner_on_event_loop(
```

Comment:
Same thing here, I think `register_job_runner` and `register_job_runner_on_event_loop` can be turned into a single function now